src/tbp/interactive/tutorials/simple_visualization/tutorial.md

```diff
@@ -0,0 +1,888 @@
+
+At this stage, running the plot should open a window with three regions and no widgets as shown below.
+
+
+![Empty interactive plot](assets/empty.png)
+
+
+## Part 2: Add an Episode Slider Widget
+
+In this part we add the first widget to the plot, the episode slider.
+We will do this in two steps.
+First, we will add a `create_widgets()` method and create a `Widget` instance.
+After that, we will implement the `WidgetOps` class that defines how the slider is created, how its state is extracted and set, and how it publishes topic messages.
+
+
+### Adding a Widget to `InteractivePlot`
+
+Start by updating the imports to include the widget wrapper and the slider.
+The rest of the file stays the same for now.
+
+```python
+from vedo import Plotter, Slider2D
+
+from tbp.interactive.widgets import (
+    VtkDebounceScheduler,
+    Widget,
+)
+```
+
+Now update `InteractivePlot.__init__` so it creates and adds widgets before showing the renderers.
+At this point, the widget will not work yet, because we have not implemented its WidgetOps class.
+
+
+```python
+class InteractivePlot:
+    """An interactive plot for a simple tutorial."""
+
+    def __init__(self, exp_path: str, data_path: str):
+        # ... same setup as Part 1 ...
+
+        self.data_parser = DataParser(exp_path)
+        self.ycb_loader = YCBMeshLoader(data_path)
+        self.event_bus = Publisher()
+        self.plotter = Plotter(shape=renderer_areas, sharecam=False).render()
+        self.scheduler = VtkDebounceScheduler(self.plotter.interactor, period_ms=33)
+
+        # NEW: Create and add widgets
+        self._widgets = self.create_widgets()
+        for w in self._widgets.values():
+            w.add()
+
+        # NEW: Set an initial value for the episode slider
+        self._widgets["episode_slider"].set_state(0)
+
+        # ... same .show(...) calls as Part 1 ...
+```
+
+Next, add a `create_widgets()` method to the class.
+This method is where we construct all widgets and decide how they are wired to the bus and scheduler.
+
+```python
+    def create_widgets(self):
+        widgets = {}
+
+        widgets["episode_slider"] = Widget[Slider2D, int](
+            widget_ops=EpisodeSliderWidgetOps(
+                plotter=self.plotter,
+                data_parser=self.data_parser,
+            ),
+            scopes = [],
+            bus=self.event_bus,
+            scheduler=self.scheduler,
+            debounce_sec=0.0,
+            dedupe=False,
+        )
+
+        return widgets
+```
+
+
+This is the first time we see the Widget wrapper.
+The Widget takes a `widget_ops` object, which defines the widget type and behavior, as well as other sharable objects like the event bus and scheduler.
+The Widget wrapper is responsible for connecting UI callbacks to the bus.
+The WidgetOps object is responsible for the widget-specific behavior, such as how to create the VTK widget and how to interpret/set its state.
```

Comment:
VTK has come up in the code, but it might be worth explicitly stating what it is, either here or earlier in the tutorial.